server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -696,6 +696,46 @@ static String getCommitCheckpointFileName(long generation) {
         return TRANSLOG_FILE_PREFIX + generation + CHECKPOINT_SUFFIX;
     }
 
+    /**
+     * Trims translog (effectively moves max visible seq# {@link Checkpoint#trimmedAboveSeqNo}) for terms below <code>belowTerm</code>
+     * and seq# above <code>aboveSeqNo</code>
+     */
+    public void trim(long belowTerm, long aboveSeqNo) throws IOException {
+        if (aboveSeqNo < 0) {
+            return;
+        }
+
+        final ChannelFactory channelFactory = FileChannel::open;
```

Comment:
nit: how much does this buy us?  I think we can just inline it and have better readability?